src/org/omegat/core/data/PluginInformation.java

```diff
@@ -42,56 +53,107 @@
+        name = findName(attrs);
+        version = findVersion(attrs, mainAttrs);
+        author = findAuthor(mainAttrs);
         description = attrs.getValue(PLUGIN_DESCRIPTION);
+        link = attrs.getValue(PLUGIN_LINK);
+        category = categoryName(attrs.getValue(PLUGIN_CATEGORY), attrs.getValue(PLUGIN_TYPE));
+        url = mu;
+        this.status = status;
     }
 
-    public PluginInformation(String className, Properties props) {
+    public PluginInformation(String className, Properties props, final String key, final URL mu, final STATUS status) {
         this.className = className;
-        name = null;
+        id = className;
+        name = className.substring(className.lastIndexOf(".") + 1);
         version = null;
         author = null;
         description = null;
+        category = categoryName(key, null);
+        link = null;
+        url = mu;
+        this.status = status;
+    }
+
+    private String categoryName(final String key1, final String key2) {
+        String key = key1 != null ? key1 : key2;
+        Optional<PluginUtils.PluginType> type = Arrays.stream(PluginUtils.PluginType.values()).filter(v ->
+                v.getTypeValue().equals(key)).findFirst();
+        if (type.isPresent()) {
+            return type.get().getTypeValue();
+        }
+        return PluginUtils.PluginType.UNKNOWN.getTypeValue();
     }
 
-    private String findName(Manifest m) {
-        Attributes attrs = m.getMainAttributes();
+    private String findId(Attributes attrs) {
+        if (attrs.getValue(PLUGIN_ID) != null) {
+            return attrs.getValue(PLUGIN_ID);
+        }
+        // fallback to className
+        return className;
+    }
+
+    private String findName(Attributes attrs) {
+
         if (attrs.getValue(PLUGIN_NAME) != null) {
             return attrs.getValue(PLUGIN_NAME);
         } else if (attrs.getValue(BUNDLE_NAME) != null) {
             return attrs.getValue(BUNDLE_NAME);
         } else if (attrs.getValue(IMPLEMENTATION_TITLE) != null) {
             return attrs.getValue(IMPLEMENTATION_TITLE);
         }
-        return null;
+        // fallback to className
+        return className.substring(className.lastIndexOf(".") + 1);
     }
 
-    private String findVersion(Manifest m) {
-        Attributes attrs = m.getMainAttributes();
+    private String findVersion(Attributes attrs, Attributes mainAttrs) {
         if (attrs.getValue(PLUGIN_VERSION) != null) {
             return attrs.getValue(PLUGIN_VERSION);
         } else if (attrs.getValue(BUNDLE_VERSION) != null) {
             return attrs.getValue(BUNDLE_VERSION);
         } else if (attrs.getValue(IMPLEMENTATION_VERSION) != null) {
             return attrs.getValue(IMPLEMENTATION_VERSION);
+        } else if (mainAttrs.getValue(PLUGIN_VERSION) != null) {
+            return mainAttrs.getValue(PLUGIN_VERSION);
+        } else if (mainAttrs.getValue(BUNDLE_VERSION) != null) {
+            return mainAttrs.getValue(BUNDLE_VERSION);
```

Comment:
Duplicated conditions should be removed.

Reply:
It is not duplicated. 
A part here is for bundled plugin or Okapi framework, that has Plugin-Version: in Main attribution and each filter has no version.